### PR TITLE
Fix #1032: Workspace doesn't bump hard enough when swiping at the end

### LINKED ISSUE #1032: Workspace doesn't bump hard enough when swiping at the end
## What Happened

If you swipe at the end of the workspace list, there is a bump animation to let you know you've reached the end and can't go any further. However, this bump is often much smaller than the bump you see when using keyboard shortcuts. Sometimes it's only a few pixels wide

## Expected Behavior

If we're bumping, we should make sure the animation is dramatic enough to be noticeable

### PR DESCRIPTION
**Pull request by JoseExposito**
Fix https://github.com/elementary/gala/issues/1032

### CODE REVIEW

File: src/WindowManager.vala [hunk 1]
```diff
@@ -105,6 +105,13 @@ namespace Gala {
          */
         public const int WORKSPACE_GAP = 24;
 
+        /**
+         * When swiping at the begin/end of the workspace list, a nudge animation is played.
+         * If using a touchpad or touchscreen the animation is smaller that using the keyboard. This
```
Comment (jlnr) [inline src/WindowManager.vala:110]: Typo?
```suggestion
         * If using a touchpad or touchscreen the animation is smaller than using the keyboard. This
```

Comment (JoseExposito) [reply src/WindowManager.vala]: Yep

**Review by danrabbit — CHANGES_REQUESTED** [review]
Hrm this seems really arbitrary and I don't think really solves the issue. This would make the nudge much large sometimes and still smaller sometimes. It wouldn't make the two nudges uniform.

When you start a workspace switch, the minimum distance the workspace is moved when the gesture is accepted is about the distance of the workspace gap, but when you start a nudge, the minimum distance is just a few pixels. So if you're swiping quickly, even with this branch you might only move 6px or so and the nudge animation is barely noticeable.

**Comment by JoseExposito** [comment]
> When you start a workspace switch, the minimum distance the workspace is moved when the gesture is accepted is about the distance of the workspace gap, but when you start a nudge, the minimum distance is just a few pixels. So if you're swiping quickly, even with this branch you might only move 6px or so and the nudge animation is barely noticeable.

Aham, now I understand what solution you want.

So, when you switch workspaces, the 100% of the movement in your touchpad matched the size of your screen.
However, in the nudge animation that 100% of movement matches 32 pixels.
I just pushed some changes so both animation are identical with the exception than the nudge animation ends at 32px.

Even though this is the "coherent" way of handling this animation it feels a little bit rough...

**Comment by hanaral** [comment]
I've already filed feature request for  - But I was wondering how difficult it would be to make the workspace bump a part of the animation before it i.e. swiping from workspace 2 to workspace 1 and continuing to pull would animate the stretch

**Comment by JoseExposito** [comment]
> I've already filed feature request for it - But I was wondering how difficult it would be to make this workspace bump a part of the animation before it i.e. swiping from workspace 2 to workspace 1 and continuing to pull would animate the stretch

I'm afraid that at the moment that'll require few changes... We'll need to handle that extra swipe movement and also change how the animation is performed drastically as, under the hood, only the current desktop + windows and the target desktop + windows are drawn.
It's possible, but it'll require a lot of work. By the way, feel free to mention me in a issue if you want to ask a question, I don't check the issue tracker that often :smile:

**Comment by danrabbit** [comment]
can confirm, this branch bumps :sunglasses:

**Comment by JoseExposito** [comment]
@danrabbit I rebased the changes in 3 commits so the change is easier to follow

**Review by danrabbit — APPROVED** [review]

**Pull request merged**